OrbitLinuxTracing/GpuTracepointEventProcessorTest.cpp

```diff
@@ -0,0 +1,589 @@
+}
+
+TEST_F(GpuTracepointEventProcessorTest, MismatchingSeqno) {
+  static constexpr pid_t kTid = 42;
+  static constexpr uint32_t kContext = 1;
+  static constexpr uint32_t kSeqno = 10;
+  static const std::string kTimeline = "timeline";
+  static constexpr uint64_t kTimestampA = 100;
+  static constexpr uint64_t kTimestampB = 200;
+  static constexpr uint64_t kTimestampD = 300;
+
+  EXPECT_CALL(mock_listener_, OnGpuJob).Times(0);
+  processor_.PushEvent(
+      MakeFakeAmdgpuCsIoctlPerfEvent(kTid, kTimestampA, kContext, kSeqno, kTimeline));
+  processor_.PushEvent(
+      MakeFakeAmdgpuSchedRunJobPerfEvent(kTimestampB, kContext, kSeqno + 1, kTimeline));
+  processor_.PushEvent(MakeFakeDmaFenceSignaledPerfEvent(kTimestampD, kContext, kSeqno, kTimeline));
+}
+
+TEST_F(GpuTracepointEventProcessorTest, MismatchingTimeline) {
+  static constexpr pid_t kTid = 42;
+  static constexpr uint32_t kContext = 1;
+  static constexpr uint32_t kSeqno = 10;
+  static const std::string kTimeline = "timeline";
+  static constexpr uint64_t kTimestampA = 100;
+  static constexpr uint64_t kTimestampB = 200;
+  static constexpr uint64_t kTimestampD = 300;
+
+  EXPECT_CALL(mock_listener_, OnGpuJob).Times(0);
+  processor_.PushEvent(
+      MakeFakeAmdgpuCsIoctlPerfEvent(kTid, kTimestampA, kContext, kSeqno, kTimeline));
+  processor_.PushEvent(
+      MakeFakeAmdgpuSchedRunJobPerfEvent(kTimestampB, kContext, kSeqno, kTimeline));
+  processor_.PushEvent(
+      MakeFakeDmaFenceSignaledPerfEvent(kTimestampD, kContext, kSeqno, kTimeline + "1"));
+}
+
+TEST_F(GpuTracepointEventProcessorTest, TwoNonOverlappingJobsDifferingByContext) {
+  static constexpr pid_t kTid = 42;
+  static constexpr uint32_t kContext1 = 1;
+  static constexpr uint32_t kContext2 = 2;
+  static constexpr uint32_t kSeqno = 10;
+  static const std::string kTimeline = "timeline";
+  static constexpr uint64_t kTimestampA1 = 100;
+  static constexpr uint64_t kTimestampB1 = 200;
+  static constexpr uint64_t kTimestampC1 = kTimestampB1;
+  static constexpr uint64_t kTimestampD1 = 300;
+  static constexpr uint64_t kNsDistanceForSameDepth = 1'000'000;
+  static constexpr uint64_t kTimestampA2 = kNsDistanceForSameDepth + 300;
+  static constexpr uint64_t kTimestampB2 = kNsDistanceForSameDepth + 400;
+  static constexpr uint64_t kTimestampC2 = kTimestampB2;
+  static constexpr uint64_t kTimestampD2 = kNsDistanceForSameDepth + 500;
+
+  orbit_grpc_protos::GpuJob expected_gpu_job1 =
+      MakeGpuJob(kTid, kContext1, kSeqno, kTimeline, 0, kTimestampA1, kTimestampB1, kTimestampC1,
+                 kTimestampD1);
+  orbit_grpc_protos::GpuJob expected_gpu_job2 =
+      MakeGpuJob(kTid, kContext2, kSeqno, kTimeline, 0, kTimestampA2, kTimestampB2, kTimestampC2,
+                 kTimestampD2);
+  orbit_grpc_protos::GpuJob actual_gpu_job1;
+  orbit_grpc_protos::GpuJob actual_gpu_job2;
+  EXPECT_CALL(mock_listener_, OnGpuJob)
```

Comment:
I am not 100% in the GpuTracepointEventProcessor, but will it guarantee in order outputs? If not consider saving the jobs in a vector, make the "expect" method returning a matcher, and use UnorderedElementsAre.

Reply:
The `GpuJob`s are created and sent to the `Listener` once all the three tracepoint events have been seen, so we can totally the order.

Reply:
Okay, thanks.